src/Controls/GlobeControls.js

```diff
@@ -521,60 +529,117 @@ class GlobeControls extends THREE.EventDispatcher {
+        panStart.copy(event.viewCoords);
+    }
 
-                panStart.copy(panEnd);
-                break;
-            case this.states.MOVE_GLOBE: {
-                const normalized = this.view.viewToNormalizedCoords(coords);
-                raycaster.setFromCamera(normalized, this.camera);
-                // If there's intersection then move globe else we stop the move
-                if (raycaster.ray.intersectSphere(pickSphere, intersection)) {
-                    normalizedIntersection.copy(intersection).normalize();
-                    moveAroundGlobe.setFromUnitVectors(normalizedIntersection, lastNormalizedIntersection);
-                    lastTimeMouseMove = Date.now();
-                } else {
-                    this.onMouseUp();
-                }
-                break; }
-            default:
+    handleRotation(event) {
+        // Stop player if needed. Player can be playing while moving mouse in the case of rotation. This is due to the
+        // fact that a damping move can occur while rotating (without the need of releasing the mouse button)
+        if (this.player.isPlaying()) {
+            this.player.stop();
         }
+        this.handlePanoramic(event);
+    }
 
-        if (this.state !== this.states.NONE) {
+    handleDrag(event) {
+        const normalized = this.view.viewToNormalizedCoords(event.viewCoords);
+
+        raycaster.setFromCamera(normalized, this.camera);
+        // If there's intersection then move globe else we stop the move
+        if (raycaster.ray.intersectSphere(pickSphere, intersection)) {
+            normalizedIntersection.copy(intersection).normalize();
+            moveAroundGlobe.setFromUnitVectors(normalizedIntersection, lastNormalizedIntersection);
+            lastTimeMouseMove = Date.now();
             this.update();
+        } else {
+            this.handleEndMovement();
+        }
+    }
+
+    handleDolly(event) {
+        dollyEnd.copy(event.viewCoords);
+        dollyDelta.subVectors(dollyEnd, dollyStart);
+
+        this.dolly(-dollyDelta.y);
+        dollyStart.copy(dollyEnd);
+
+        this.update();
+    }
+
+    handlePan(event) {
+        panEnd.copy(event.viewCoords);
+        panDelta.subVectors(panEnd, panStart);
+        this.mouseToPan(panDelta.x, panDelta.y);
+        panStart.copy(panEnd);
+
+        this.update();
+    }
+
+    handlePanoramic(event) {
+        rotateEnd.copy(event.viewCoords);
+        rotateDelta.subVectors(rotateEnd, rotateStart);
+
+        const gfx = this.view.mainLoop.gfxEngine;
+
+        sphericalDelta.theta -= 2 * Math.PI * rotateDelta.x / gfx.width * this.rotateSpeed;
+        // rotating up and down along whole screen attempts to go 360, but limited to 180
+        sphericalDelta.phi -= 2 * Math.PI * rotateDelta.y / gfx.height * this.rotateSpeed;
+
+        rotateStart.copy(rotateEnd);
+        this.update();
+    }
+
+    handleEndMovement() {
+        if (this.enabled === false) { return; }
```

Comment:
could you move `this.enable` to `stateControls`?

Reply:
I moved it.